node/src/neighborhood/node_location.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use ip_country_lib;
+use ip_country_lib::country_finder::{CountryCodeFinder, COUNTRY_CODE_FINDER};
+use std::net::IpAddr;
+
+#[allow(dead_code)]
+#[derive(Clone, Debug, Default)]
+pub struct NodeLocation {
+    pub(crate) country_code: String,
+    pub(crate) free_world_bit: bool,
+}
+
+impl PartialEq<Self> for NodeLocation {
+    fn eq(&self, other: &Self) -> bool {
+        self.country_code == other.country_code
+    }
+}
+
+impl Eq for NodeLocation {}
+
+pub fn get_node_location(ip: Option<IpAddr>) -> Option<NodeLocation> {
+    match ip {
+        Some(ip_addr) => {
+            let country = CountryCodeFinder::find_country(&COUNTRY_CODE_FINDER, ip_addr);
+            match country {
+                Some(country) => Some(NodeLocation {
+                    country_code: country.iso3166.to_string(),
+                    free_world_bit: country.free_world,
+                }),
+                None => None,
+            }
+        }
+        None => None,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::neighborhood::gossip::GossipBuilder;
+    use crate::neighborhood::node_location::{get_node_location, NodeLocation};
+    use crate::neighborhood::node_record::{NodeRecord, NodeRecordMetadata};
+    use crate::sub_lib::node_addr::NodeAddr;
+    use crate::test_utils::neighborhood_test_utils::{
+        db_from_node, make_node_record, pick_country_code_record,
+    };
+    use std::net::{IpAddr, Ipv4Addr};
+
+    #[test]
+    fn test_node_location() {
+        let node_location =
+            get_node_location(Some(IpAddr::V4(Ipv4Addr::new(125, 125, 125, 1)))).unwrap();
+
+        assert_eq!(node_location.country_code, "CN");
+        assert_eq!(node_location.free_world_bit, false);
+    }
+
+    #[test]
+    fn construct_node_record_metadata_with_free_world_bit() {
```

Comment:
This whole test, if broken down to its pieces, doesn't really make any sense after the previous test. It performs exactly the same kind of assertions. 

Look what you did:

1) you create an object of NodeRecordMetadata (which you aren't gonna wholly inspect in the test; and you probably even don't need to as you don't need this whole test),

2) you **inject** a specific node_location generated **the same way** as you did in the previous test. 

3) you go on by taking this modified metadata structure and applying assertions regarding only the node location aspect of it. However, in the end, that's nothing extra compared to what you did one test before. Plus the first assertion is doubled by the two next. 

I think this test should be wholly deleted.